js/framework/AnalyticsConstants.js

```diff
@@ -561,12 +561,13 @@ if (!OO.Analytics.EVENT_DATA)
   EVENT_DATA.VideoBitrateProfileLookupData = function(bitrateProfileArray)
   {
     var checkBitrateProfileList = OO._.bind(checkDataType, this, "VideoBitrateProfileLookupData");
-    var list = checkBitrateProfileList(bitrateProfileArray, "bitrateProfileArray", ["array"]);
+    var list = checkBitrateProfileList(bitrateProfileArray, "bitrateProfileArray", ["array"]) || [];
     this.profiles = {};
-    for(var key in list)
-    {
-      var entry = list[key];
-      this.profiles[entry.id] = entry;
+    for (var i = 0; i < list.length; i++) {
```

Comment:
Sorry to nitpick, but can you please place the `{`'s on a newline. We have been trying to adopt this style for our new code.